crates/rspack_plugin_copy/src/lib.rs

```diff
@@ -789,6 +818,29 @@ fn set_info(target: &mut AssetInfo, info: Info) {
   }
 }
 
+async fn handle_transform(
+  transformer: &TransformerFn,
+  source_vec: Vec<u8>,
+  absolute_filename: Utf8PathBuf,
+  source: &mut RawSource,
+  diagnostics: &Mutex<Vec<Diagnostic>>,
```

Comment:
`&Mutex<Vec<Diagnostics>>` is super weird in async context, `Arc<Mutex` seems better, can you help fix this